### DISCUSSION TITLE
Code block command

### DISCUSSION THREAD

**Discussion by osmiumforge** [Taitava/obsidian-shellcommands#370] [Ideas]
A useful feature would be the ability to run a command on the contents of a fenced code block (based on the language set) and render the output, so that the input can be edited and automatically updated

**Comment by Taitava** [comment]
Thank you for your suggestion! 🙂 

The way I see this, there could be a new [`{{variable}}`](https://publish.obsidian.md/shellcommands/Variables/Variables+-+general+principles) named e.g. `{{code_block}}`, which could have two choices:
- `{{code_block:content}}`: Get the actual code of the first fenced code block on the currently active note file.
- `{{code_block:language}}` Get whatever text part is placed at the first line after ` ``` `. (Or: `{{code_block:type}}`)

For example, you could have this kind of note file:
````markdown
My note content before code.

```bash
echo "Hello world!"
```

Note content after code.
````

And you could have a shell command like this:

```bash
{{code_block:language}} -c {{code_block:content}}
```

When executed, it would become:
```bash
bash -c \"Hello\ world\!\"
```

**Reply by Taitava** [reply]
Anyway, there are some other plugins that execute code blocks:
- https://obsidian.md/plugins?id=code-emitter
- https://obsidian.md/plugins?id=snippets

I haven't tried these myself.

**Reply by Taitava** [reply]
> and render the output

Hmm, I do see the usefulnes of having an output element next to the code block. It's just that there might be quite a lot of work to achieve that. Perhaps something like this could work: (this just demonstrates an idea, it's **not** implemented)

````markdown
My note content before code.

```bash
echo "Hello world!"
```

```shell-commands-output
AFancyNameForMyOutputContainer
```

````

Then in a shell command's settings, one could define that `stdout`/`stderr` (or both) output should go to an _Output container_ named `AFancyNameForMyOutputContainer` in the currently active file.

Need to consider if the output should be either:
 - permanently stored (i.e. could be seen again after restarting Obsidian without re-executing the shell command. This would modify the note file's content), **or**
 - temporarily visible. Closing Obsidian would discard the output. Good if user doesn't want their note file to be tampered with.
